confidential-datahub/secret/src/secret/mod.rs

```diff
@@ -24,12 +26,20 @@ pub struct Secret {
     pub r#type: SecretContent,
 }
 
+impl Secret {
+    pub async fn unseal(&self) -> Result<Vec<u8>> {
+        match &self.r#type {
+            SecretContent::Envelope(env) => env.unseal().await,
+            SecretContent::Vault(_) => todo!(),
+        }
```

Comment:
What about checking the `version` here? I think it's a good field to have. It looks to me like `unseal_secret` will deserialize everything, create the `Secret` struct, and then invoke `unseal`. But you could end up with all the right fields but a subtle difference due to version change, in which case you would want to catch it.

Reply:
Yes, thanks! Let me fix this